### PR TITLE
Bug fixes

### PR DESCRIPTION
**Pull request by Shubha-accenture**
1. Implemented server side pagination
2. Pagination changes for fetching first and last page
3. Vertical spacing for Dataset explorer (Schema and Preview)
4. Addressed review comments on dataset explorer

### CODE REVIEW

**Review by ojarjur — COMMENTED** [review]
Looks good on the Python front, but we need someone else to review the Typescript changes before merging.

File: dataproc_jupyter_plugin/utils/utils.py
Comment (ojarjur) [inline dataproc_jupyter_plugin/utils/utils.py:1]: We should not have any files named just `utils`. That's never going to be the right name to for a package in this repo.

File: dataproc_jupyter_plugin/utils/credentials.py
Comment (ojarjur) [inline dataproc_jupyter_plugin/utils/credentials.py:1]: Please rename this to `dataproc_juptyer_plugin/credentials.py`.

There's no reason to have anything in the codebase named `utils`.

Comment (harsha-accenture) [reply dataproc_jupyter_plugin/utils/credentials.py:1]: 1. credentials.py is renamed to gcloudOperations.py
2. utils folder is renamed to commons as it contains all the common code.

File: src/bigQuery/previewDataInfo.tsx
Comment (ptwng) [inline src/bigQuery/previewDataInfo.tsx:26]: this wouldn't work if the user resizes the browser after js was loaded

**Review by ptwng — APPROVED** [review]

**Comment by ptwng** [comment]
Just the one thing around statically grabbing the window width

**Review by ptwng — CHANGES_REQUESTED** [review]
Just the one thing around statically grabbing the screen width

**Review by ojarjur — COMMENTED** [review]
The Python changes are fine, so this can be merged once Pat approves the TS changes

**Review by ptwng — APPROVED** [review]

**Pull request merged**
